expect_test.go

```diff
@@ -341,6 +341,99 @@ func TestExpect_Branches(t *testing.T) {
 	e5.chain.assertFlags(t, 0)
 }
 
+func TestExpect_ErrorPropagation(t *testing.T) {
+	t.Run("subsequent operations", func(t *testing.T) {
+		ctr := 0
+		reporter := newMockReporter(t)
+		reporter.reportCb = func() {
+			ctr++
+		}
+
+		arr := NewArray(reporter, []interface{}{"foo"})
+		arr.IsEmpty().IsEmpty() // The second .IsEmpty() call does not report failure
+
+		arr.chain.assertFlags(t, flagFailed)
+		assert.Equal(t, true, reporter.reported)
+		assert.Equal(t, 1, ctr)
+	})
+
+	t.Run("newly created child objects", func(t *testing.T) {
```

Comment:
nit: for clarity, let's use `child` or `children` instead of `child object(s)` here and in other tests.